site/src/request_handlers/github.rs

```diff
@@ -118,26 +116,25 @@ async fn handle_rust_timer(
         return Ok(github::Response);
     }
 
-    if let Some(captures) = BODY_TIMER_QUEUE.captures(&comment.body) {
-        let include = captures.get(1).map(|v| v.as_str());
-        let exclude = captures.get(2).map(|v| v.as_str());
-        let runs = captures.get(3).and_then(|v| v.as_str().parse::<i32>().ok());
-        {
-            let conn = ctxt.conn().await;
-            conn.queue_pr(issue.number, include, exclude, runs).await;
-        }
-        main_client
-            .post_comment(
-                issue.number,
+    if let Some(queue) = parse_queue_command(&comment.body) {
+        let msg = match queue {
+            Ok(cmd) => {
+                let conn = ctxt.conn().await;
+                conn.queue_pr(issue.number, cmd.include, cmd.exclude, cmd.runs)
+                    .await;
                 format!(
                     "Awaiting bors try build completion.
 
 @rustbot label: +S-waiting-on-perf
 
 {COMMENT_MARK_TEMPORARY}"
-                ),
-            )
-            .await;
+                )
+            }
+            Err(error) => {
+                format!("Error occurred while parsing comment: {error}")
+            }
+        };
+        main_client.post_comment(issue.number, msg).await;
```

Comment:
do we need to take special care (like sanitization) in posting the error message to GH as its text will be partially controlled by users' input commands?

Reply:
That's an interesting idea. I don't think so, it should have the same rules as when you create the comment manually. I guess that the user could make the bot print "Unknown command argument SEND MONEY TO THE FOLLOWING BITCOIN ADDRESS TO SPONSOR THE RUST FOUNDATION", but I don't think that they can XSS or something like that (that would be really bad hole in GH's security).